api/types.go

```diff
@@ -631,6 +704,23 @@ func (po *PinOptions) FromQuery(q url.Values) error {
 	return nil
 }
 
+// PinDepth indicates how deep a pin should be pinned, with
+// -1 meaning "to the bottom", or "recursive".
+type PinDepth int
+
+// ToPinMode converts PinDepth to PinMode
+func (pd PinDepth) ToPinMode() PinMode {
+	switch pd {
+	case -1:
+		return PinModeRecursive
+	case 0:
+		return PinModeDirect
+	default:
+		logger.Warnf("bad pin depth: %d", pd)
+		return PinModeRecursive
+	}
+}
```

Comment:
Since PinMode and Depth are quite related, we need easy conversion between the two. Therefore PinDepth can use it's own type.